From 291865bff36306d70c6975e73eb299d977abbc2c Mon Sep 17 00:00:00 2001
From: Valentine Barshak <valentine.barshak@cogentembedded.com>
Date: Tue, 5 Nov 2019 19:40:06 +0300
Subject: [PATCH] arm64: dts: renesas: ulcb-vb: Fix lvds0 port routing

This fixes LVDS port routing on all H3ULCB VideoBox
boards which is needed for the FDPLink output.

Signed-off-by: Valentine Barshak <valentine.barshak@cogentembedded.com>
---
 arch/arm64/boot/dts/renesas/r8a7795-es1-h3ulcb-vb.dts     | 11 +++++++++--
 arch/arm64/boot/dts/renesas/r8a7795-es1-h3ulcb-vb2.dts    | 11 +++++++++--
 arch/arm64/boot/dts/renesas/r8a7795-es1-h3ulcb-vbm.dts    |  8 +++++---
 arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-4x2g-vb.dts    | 11 +++++++++--
 arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-4x2g-vb2.1.dts | 11 +++++++++--
 arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-4x2g-vb2.dts   | 11 +++++++++--
 arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-4x2g-vbm.dts   |  8 +++++---
 arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-vb.dts         | 11 +++++++++--
 arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-vb2.1.dts      | 11 +++++++++--
 arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-vb2.dts        | 11 +++++++++--
 arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-vbm.dts        |  8 +++++---
 arch/arm64/boot/dts/renesas/ulcb-vb.dtsi                  |  2 +-
 arch/arm64/boot/dts/renesas/ulcb-vb2.dtsi                 |  2 +-
 arch/arm64/boot/dts/renesas/ulcb-vbm.dtsi                 |  2 +-
 14 files changed, 90 insertions(+), 28 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a7795-es1-h3ulcb-vb.dts b/arch/arm64/boot/dts/renesas/r8a7795-es1-h3ulcb-vb.dts
index db9f80f..310d2df 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795-es1-h3ulcb-vb.dts
+++ b/arch/arm64/boot/dts/renesas/r8a7795-es1-h3ulcb-vb.dts
@@ -34,8 +34,15 @@
 				remote-endpoint = <&rcar_dw_hdmi1_in>;
 			};
 		};
-		port@3 {
-			endpoint {
+	};
+};
+
+&lvds0 {
+	status = "okay";
+
+	ports {
+		port@1 {
+			lvds0_out: endpoint {
 				remote-endpoint = <&lvds_in>;
 			};
 		};
diff --git a/arch/arm64/boot/dts/renesas/r8a7795-es1-h3ulcb-vb2.dts b/arch/arm64/boot/dts/renesas/r8a7795-es1-h3ulcb-vb2.dts
index 6d51ffd..65b9aa5 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795-es1-h3ulcb-vb2.dts
+++ b/arch/arm64/boot/dts/renesas/r8a7795-es1-h3ulcb-vb2.dts
@@ -40,8 +40,15 @@
 				remote-endpoint = <&rcar_dw_hdmi1_in>;
 			};
 		};
-		port@3 {
-			endpoint {
+	};
+};
+
+&lvds0 {
+	status = "okay";
+
+	ports {
+		port@1 {
+			lvds0_out: endpoint {
 				remote-endpoint = <&lvds_in>;
 			};
 		};
diff --git a/arch/arm64/boot/dts/renesas/r8a7795-es1-h3ulcb-vbm.dts b/arch/arm64/boot/dts/renesas/r8a7795-es1-h3ulcb-vbm.dts
index 64815d4..a3b51da 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795-es1-h3ulcb-vbm.dts
+++ b/arch/arm64/boot/dts/renesas/r8a7795-es1-h3ulcb-vbm.dts
@@ -15,10 +15,12 @@
 	model = "Renesas H3ULCB Videobox Mini board based on r8a7795";
 };
 
-&du {
+&lvds0 {
+	status = "okay";
+
 	ports {
-		port@3 {
-			endpoint {
+		port@1 {
+			lvds0_out: endpoint {
 				remote-endpoint = <&lvds_in>;
 			};
 		};
diff --git a/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-4x2g-vb.dts b/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-4x2g-vb.dts
index ce16cab..1f8c229 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-4x2g-vb.dts
+++ b/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-4x2g-vb.dts
@@ -33,8 +33,15 @@
 				remote-endpoint = <&rcar_dw_hdmi1_in>;
 			};
 		};
-		port@3 {
-			endpoint {
+	};
+};
+
+&lvds0 {
+	status = "okay";
+
+	ports {
+		port@1 {
+			lvds0_out: endpoint {
 				remote-endpoint = <&lvds_in>;
 			};
 		};
diff --git a/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-4x2g-vb2.1.dts b/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-4x2g-vb2.1.dts
index 1a9d0be..aa99875 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-4x2g-vb2.1.dts
+++ b/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-4x2g-vb2.1.dts
@@ -33,8 +33,15 @@
 				remote-endpoint = <&rcar_dw_hdmi1_in>;
 			};
 		};
-		port@3 {
-			endpoint {
+	};
+};
+
+&lvds0 {
+	status = "okay";
+
+	ports {
+		port@1 {
+			lvds0_out: endpoint {
 				remote-endpoint = <&lvds_in>;
 			};
 		};
diff --git a/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-4x2g-vb2.dts b/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-4x2g-vb2.dts
index 1351c6f..fb45133 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-4x2g-vb2.dts
+++ b/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-4x2g-vb2.dts
@@ -33,8 +33,15 @@
 				remote-endpoint = <&rcar_dw_hdmi1_in>;
 			};
 		};
-		port@3 {
-			endpoint {
+	};
+};
+
+&lvds0 {
+	status = "okay";
+
+	ports {
+		port@1 {
+			lvds0_out: endpoint {
 				remote-endpoint = <&lvds_in>;
 			};
 		};
diff --git a/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-4x2g-vbm.dts b/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-4x2g-vbm.dts
index 352cc0d..b415e88 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-4x2g-vbm.dts
+++ b/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-4x2g-vbm.dts
@@ -15,10 +15,12 @@
 	model = "Renesas H3ULCB with 8GiB (4 x 2 GiB) Videobox Mini board based on r8a7795";
 };
 
-&du {
+&lvds0 {
+	status = "okay";
+
 	ports {
-		port@3 {
-			endpoint {
+		port@1 {
+			lvds0_out: endpoint {
 				remote-endpoint = <&lvds_in>;
 			};
 		};
diff --git a/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-vb.dts b/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-vb.dts
index 26c15f4..9517ff0 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-vb.dts
+++ b/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-vb.dts
@@ -33,8 +33,15 @@
 				remote-endpoint = <&rcar_dw_hdmi1_in>;
 			};
 		};
-		port@3 {
-			endpoint {
+	};
+};
+
+&lvds0 {
+	status = "okay";
+
+	ports {
+		port@1 {
+			lvds0_out: endpoint {
 				remote-endpoint = <&lvds_in>;
 			};
 		};
diff --git a/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-vb2.1.dts b/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-vb2.1.dts
index 43c20b0..5f97b60 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-vb2.1.dts
+++ b/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-vb2.1.dts
@@ -33,8 +33,15 @@
 				remote-endpoint = <&rcar_dw_hdmi1_in>;
 			};
 		};
-		port@3 {
-			endpoint {
+	};
+};
+
+&lvds0 {
+	status = "okay";
+
+	ports {
+		port@1 {
+			lvds0_out: endpoint {
 				remote-endpoint = <&lvds_in>;
 			};
 		};
diff --git a/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-vb2.dts b/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-vb2.dts
index 816c7da..56d43d07 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-vb2.dts
+++ b/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-vb2.dts
@@ -33,8 +33,15 @@
 				remote-endpoint = <&rcar_dw_hdmi1_in>;
 			};
 		};
-		port@3 {
-			endpoint {
+	};
+};
+
+&lvds0 {
+	status = "okay";
+
+	ports {
+		port@1 {
+			lvds0_out: endpoint {
 				remote-endpoint = <&lvds_in>;
 			};
 		};
diff --git a/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-vbm.dts b/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-vbm.dts
index 053a60e..7fcf527 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-vbm.dts
+++ b/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb-vbm.dts
@@ -15,10 +15,12 @@
 	model = "Renesas H3ULCB Videobox Mini board based on r8a7795";
 };
 
-&du {
+&lvds0 {
+	status = "okay";
+
 	ports {
-		port@3 {
-			endpoint {
+		port@1 {
+			lvds0_out: endpoint {
 				remote-endpoint = <&lvds_in>;
 			};
 		};
diff --git a/arch/arm64/boot/dts/renesas/ulcb-vb.dtsi b/arch/arm64/boot/dts/renesas/ulcb-vb.dtsi
index 67903db..6fb2b38 100644
--- a/arch/arm64/boot/dts/renesas/ulcb-vb.dtsi
+++ b/arch/arm64/boot/dts/renesas/ulcb-vb.dtsi
@@ -145,7 +145,7 @@
 
 		port {
 			lvds_in: endpoint {
-				remote-endpoint = <&du_out_lvds0>;
+				remote-endpoint = <&lvds0_out>;
 			};
 		};
 	};
diff --git a/arch/arm64/boot/dts/renesas/ulcb-vb2.dtsi b/arch/arm64/boot/dts/renesas/ulcb-vb2.dtsi
index 50cdfd8..1a73059 100644
--- a/arch/arm64/boot/dts/renesas/ulcb-vb2.dtsi
+++ b/arch/arm64/boot/dts/renesas/ulcb-vb2.dtsi
@@ -137,7 +137,7 @@
 
 		port {
 			lvds_in: endpoint {
-				remote-endpoint = <&du_out_lvds0>;
+				remote-endpoint = <&lvds0_out>;
 			};
 		};
 	};
diff --git a/arch/arm64/boot/dts/renesas/ulcb-vbm.dtsi b/arch/arm64/boot/dts/renesas/ulcb-vbm.dtsi
index 067607a..b7e3817 100644
--- a/arch/arm64/boot/dts/renesas/ulcb-vbm.dtsi
+++ b/arch/arm64/boot/dts/renesas/ulcb-vbm.dtsi
@@ -36,7 +36,7 @@
 
 		port {
 			lvds_in: endpoint {
-				remote-endpoint = <&du_out_lvds0>;
+				remote-endpoint = <&lvds0_out>;
 			};
 		};
 	};
-- 
2.7.4

